lib/ansible/modules/cloud/azure/azure_rm_sql_server.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_sql_server
+version_added: "2.5"
+short_description: Manage an Server.
+description:
+    - Create, update and delete an instance of Server.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group that contains the resource. You can obtain this value from the Azure Resource Manager API or the portal.
+        required: True
+    name:
+        description:
+            - The name of the server.
+        required: True
+    location:
+        description:
+            - Resource location.
+        required: True
+    tags:
+        description:
+            - Resource tags.
+        required: False
+    identity:
+        description:
+            - The Azure Active Directory identity of the server.
+        required: False
+        suboptions:
+            type:
+                description:
+                    - "The identity type. Set this to 'SystemAssigned' in order to automatically create and assign an Azure Active Directory principal for th
+                       e resource. Possible values include: 'SystemAssigned'"
+                required: False
+    administrator_login:
+        description:
+            - Administrator username for the server. Once created it cannot be changed.
+        required: False
+    administrator_login_password:
+        description:
+            - The administrator login password (required for server creation).
+        required: False
+    version:
+        description:
+            - The version of the server.
+        required: False
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+      - name: Create (or update) Servers
+        azure_rm_sql_server:
+          resource_group: "{{ resource_group }}"
+          name: zims-server
+          location: westus
+          tags: "{{ tags }}"
+          identity:
+            type: "{{ type }}"
+          administrator_login: mylogin
+          administrator_login_password: Testpasswordxyz12!
+          version: "{{ version }}"
+'''
+
+RETURN = '''
+state:
```

Comment:
We should start looking at stopping this behaviour and returning relevant values in the root level of the return. When testing this module I see the following is returned

```
"state": {
    "administrator_login": "username",
    "fully_qualified_domain_name": "jborean-sql-server.database.windows.net",
    "id": "/subscriptions/3f7e29ba-24e0-42f6-8d9c-5149a14bda37/resourceGroups/jborean-testgroup/providers/Microsoft.Sql/servers/jborean-sql-server",
    "kind": "v12.0",
    "location": "australiaeast",
    "name": "jborean-sql-server",
    "state": "Ready",
    "type": "Microsoft.Sql/servers",
    "version": "12.0"
}
```

So let's do the following and create a return value for:
* fqdn
* id
* version

I don't think the other values are really necessary and are more suited towards a facts module if we want to gather that information. I decided to omit the `location`, `admin login`, `kind, `name`, `state` and `type` as they would be ones you set in the module parameters and already know. Version is a tiny bit different as you can omit it and I could argue `type` could potentially be returned if you can specify different types in the future.